### PR TITLE
Update PoissonNLLLoss document. ;test=docs_preview

### PR DESCRIPTION
**Pull request by LyndonKong**
### PR types
New features

### Chnages
Add docs for PoissonNLLLoss and poisson_nll_loss

### References
code and doc string link: https://github.com/PaddlePaddle/Paddle/pull/51117
rfc link: https://github.com/PaddlePaddle/community/pull/395

### CODE REVIEW

File: docs/api/paddle/nn/PoissonNLLLoss_cn.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+损失函数中的最后一项可以使用Stirling公式近似，该近似项的计算公式为
+    .. math::
+        \text{label} * \log(\text{label}) - \text{label} + 0.5 * \log(2 * \pi * \text{label})
+将label和每个元素都为1的同样形状的张量比较，对label值超过1的索引处考虑此项近似，对label的值小于等于1的索引处设置此项近似为0进行遮盖。
+
+参数
+:::::::::
+    - **log_input** (bool，可选) - 输入是否为对数函数映射后结果，默认值为 ``True``。
+    - **full** (bool，可选) - 是否在损失计算中包括Stirling近似项。默认值为 ``False``。
+    - **epsilon** (float，可选) - 在 :attr:`log_input`为 ``True`` 时使用的常数小量。默认值为1e-8。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/PoissonNLLLoss_cn.rst]: 注意:attr:`log_input` 后加空格
image

File: docs/api/paddle/nn/PoissonNLLLoss_cn.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - **reduction** (str，可选) - 指定应用于输出结果的计算方式，可选值有 ``none``、``mean`` 和 ``sum``。默认为 ``mean``，计算 ``mini-batch`` loss 均值。设置为 `sum` 时，计算 `mini-batch` loss 的总和。设置为 ``none`` 时，则返回 loss Tensor。默认值下为 ``mean``。
+    - **name** (str，可选) - 具体用法请参见 :ref:`api_guide_Name`，一般无需设置，默认值为 None。
+
+形状
+:::::::::
+    - **input** (Tensor) - 输入 `Tensor`，其形状为 :math::`[N, *]`，其中 :math:`*` 表示任何数量的额外维度。数据类型为 float32 或 float64。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/PoissonNLLLoss_cn.rst]: 请修复
image

File: docs/api/paddle/nn/PoissonNLLLoss_cn.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
+.. _cn_api_nn_loss_PoissonNLLLoss:
+
+PoissonNLLLoss
+-------------------------------
+
+.. py:class:: paddle.nn.PoissonNLLLoss(log_input=False, full=False, epsilon=1e-8, reduction='mean', name=None)
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/PoissonNLLLoss_cn.rst]: 预览有错，应该都是空行的问题，请严格参考文档写作规范提供的模板格式来写
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_docs_guidelines_cn.html#zhongwenmuban
image

File: docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    - **name** (str，可选) - 具体用法请参见 :ref:`api_guide_Name`，一般无需设置，默认值为 None。
+
+返回
+:::::::::
+`Tensor`，返回存储表示 `poisson negative log likelihood loss` 的损失值。
+
+代码示例
+:::::::::
+
+COPY-FROM: paddle.nn.functional.nll_loss
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst]: copy from的标签错了吧，不是 nll_loss

File: docs/api/paddle/nn/PoissonNLLLoss_cn.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    .. math::
+        \text{label} * \log(\text{label}) - \text{label} + 0.5 * \log(2 * \pi * \text{label})
+将label和每个元素都为1的同样形状的张量比较，对label值超过1的索引处考虑此项近似，对label的值小于等于1的索引处设置此项近似为0进行遮盖。
+
+参数
+:::::::::
+    - **log_input** (bool，可选) - 输入是否为对数函数映射后结果，默认值为 ``True``。
+    - **full** (bool，可选) - 是否在损失计算中包括Stirling近似项。默认值为 ``False``。
+    - **epsilon** (float，可选) - 在 :attr:`log_input`为 ``True`` 时使用的常数小量。默认值为1e-8。
+    - **reduction** (str，可选) - 指定应用于输出结果的计算方式，可选值有 ``none``、``mean`` 和 ``sum``。默认为 ``mean``，计算 ``mini-batch`` loss 均值。设置为 `sum` 时，计算 `mini-batch` loss 的总和。设置为 ``none`` 时，则返回 loss Tensor。默认值下为 ``mean``。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/PoissonNLLLoss_cn.rst]: mini-batch 前后的引用方式不一样？
image

File: docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+.. py:function:: paddle.nn.functional.poisson_nll_loss(input, label, log_input=False, full=False, epsilon=1e-8, reduction='mean', name=None)
+
+返回 `poisson negative log likelihood`。可在 :ref:`cn_api_nn_loss_PoissonNLLLoss` 查看详情。
+
+参数
+:::::::::
+    - **input** (Tensor) - 输入 :attr:`Tensor`，对应泊松分布的期望，其形状为 :math:`[N, *]`，其中 :math:`*` 表示任何数量的额外维度。数据类型为 float32 或 float64。
+    - **label** (Tensor) - 标签 :attr:`Tensor`， 形状、数据类型和 :attr:`input` 相同。
+    - **log_input** (bool，可选) - 输入是否为对数函数映射后结果，如果为 ``True``，则 loss 当中第一项的计算公式为
+        .. math::
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst]: 请参考模板来写，将以下进行修复
image

**Review by sunzhongkai588 — COMMENTED** [review]
除提到的错误外，还需修改：
- 在paddle.nn.overview的文档中增加该api描述 https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/nn/Overview_cn.html#loss
- 在api_label中增加api和标签https://github.com/PaddlePaddle/docs/blob/develop/docs/api/api_label

**Comment by LyndonKong** [comment]
update document and ready to be reviewed.

File: docs/api/paddle/nn/Overview_cn.rst [hunk 7]
```diff
@@ -258,6 +258,7 @@ Loss 层
     " :ref:`paddle.nn.MarginRankingLoss <cn_api_nn_loss_MarginRankingLoss>` ", "MarginRankingLoss 层"
     " :ref:`paddle.nn.MSELoss <cn_api_paddle_nn_MSELoss>` ", "均方差误差损失层"
     " :ref:`paddle.nn.NLLLoss <cn_api_nn_loss_NLLLoss>` ", "NLLLoss 层"
+    " :ref:`paddle.nn.PoissonNLLLoss <cn_api_nn_loss_PoissonNLLLoss>`", "PoissonNLLLoss 层"
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/Overview_cn.rst]: 引用的标签应当和 api/label 、api文档第一行的标签一致，目前`_api_nn_loss_PoissonNLLLoss` 和 api/label中的 `_api_paddle_nn_PoissonNLLLoss` 不一致

- 建议这里修改为 `cn_api_paddle_nn_PoissonNLLLoss`
- 同时，相关api文档第一行也修改为 `.. _cn_api_paddle_nn_PoissonNLLLoss:`

File: docs/api/paddle/nn/PoissonNLLLoss_cn.rst [hunk 8]
```diff
@@ -0,0 +1,53 @@
+.. _cn_api_nn_loss_PoissonNLLLoss:
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/PoissonNLLLoss_cn.rst]: 如上所诉，标签需要保持一致，第一行也修改为` .. _cn_api_paddle_nn_PoissonNLLLoss:`

File: docs/api/paddle/nn/Overview_cn.rst [hunk 9]
```diff
@@ -476,6 +477,7 @@ Embedding 相关函数
     " :ref:`paddle.nn.functional.dice_loss <cn_api_fluid_layers_dice_loss>` ", "用于比较预测结果跟标签之间的相似度"
     " :ref:`paddle.nn.functional.hsigmoid_loss <cn_api_nn_functional_hsigmoid_loss>` ", "层次 sigmoid 损失函数"
     " :ref:`paddle.nn.functional.l1_loss <cn_paddle_nn_functional_loss_l1>` ", "用于计算 L1 损失"
+    " :ref:`paddle.nn.functional.poisson_nll_loss <cn_paddle_nn_functional_loss_poisson_nll_loss>` ", "用于计算泊松分布的负对数似然损失"
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/Overview_cn.rst]: 引用的标签应当和 api/label 、api文档第一行的标签一致，目前`poisson_nll_loss` 的标签在者三个地方都不统一

- 建议和 api/label 中的标签保持一致，即改成 `cn_api_paddle_nn_functional_poisson_nll_loss`
- 同时，相关api文档第一行也修改为 `.. _cn_api_paddle_nn_functional_poisson_nll_loss:`

File: docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst [hunk 10]
```diff
@@ -0,0 +1,43 @@
+.. _cn_api_nn_functional_poisson_nll_loss:
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst]: 如上所诉，标签需要保持一致，第一行也修改为 `.. _cn_api_paddle_nn_functional_poisson_nll_loss:`

File: docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+poisson_nll_los
+-------------------------------
+.. py:function:: paddle.nn.functional.poisson_nll_loss(input, label, log_input=False, full=False, eps=1e-8, reduction='mean', name=None)
+
+返回 `poisson negative log likelihood`。可在 :ref:`cn_api_nn_loss_PoissonNLLLoss` 查看详情。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst]: 同理，如果之前标签修改了，这部分引用也进行同步，即改成 cn_api_paddle_nn_PoissonNLLLoss

File: docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+        \text{input} - \text{label} * \log(\text{input}+\text{eps})
+
+    其中 :attr:`eps` 为数值稳定使用的常数小量。
+    如果为 ``False``，则 loss 的计算公式为
+
+    .. math::
+        \exp(\text{input}) - \text{label} * \text{input}
+
+    默认值为 ``True``。
+    - **full** (bool，可选) - 是否在损失计算中包括Stirling近似项。该近似项的计算公式为
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst]: full 参数描述之前空一行吧
image

**Comment by LyndonKong** [comment]
The above issues are fixed in our new commit.

File: docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst [hunk 13]
```diff
@@ -0,0 +1,44 @@
+.. _cn_api_paddle_nn_functional_poisson_nll_loss:
+
+poisson_nll_los
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/poisson_nll_loss_cn.rst]: 是不是少了s

File: docs/api/paddle/nn/PoissonNLLLoss_cn.rst [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+:::::::::
+    - **log_input** (bool，可选) - 输入是否为对数函数映射后结果，默认值为 ``True``。
+    - **full** (bool，可选) - 是否在损失计算中包括Stirling近似项。默认值为 ``False``。
+    - **eps** (float，可选) - 在 :attr:`log_input` 为 ``True`` 时使用的常数小量。默认值为1e-8。
+    - **reduction** (str，可选) - 指定应用于输出结果的计算方式，可选值有 ``none``、``mean`` 和 ``sum``。默认为 ``mean``，计算 ``mini-batch`` loss 均值。设置为 ``sum`` 时，计算 ``mini-batch`` loss 的总和。设置为 ``none`` 时，则返回 loss Tensor。默认值下为 ``mean``。
+    - **name** (str，可选) - 具体用法请参见 :ref:`api_guide_Name` ，一般无需设置，默认值为 None。
+
+形状
+:::::::::
+    - **input** (Tensor) - 输入 `Tensor`，其形状为 :math::`[N, *]` ，其中 :math:`*` 表示任何数量的额外维度。数据类型为 float32 或 float64。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/PoissonNLLLoss_cn.rst]: :math::`[N, *]` 多了个冒号
image

File: docs/api/paddle/nn/PoissonNLLLoss_cn.rst [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+.. math::
+    \text{loss}(\text{input}, \text{label}) = \text{input} - \text{label} * \log(\text{input}+\text{eps}) + \log(\text{label!})
+
+其中 `eps` 是 ``True`` 时使用的常数小量，使得 loss 计算过程中不会导致对0求对数情况的出现。
+当 `log_input` 设置为 `False` 时，损失函数的数学计算公式为：
+
+.. math::
+    \text{loss}(\text{input}, \text{label}) = \exp(\text{input}) - \text{label} * \text{input} + \log(\text{label!})
+
+损失函数中的最后一项可以使用Stirling公式近似，该近似项的计算公式为
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/PoissonNLLLoss_cn.rst]: stirling 左右加空格（zhong英文间加空格）

image

**Review by sunzhongkai588 — COMMENTED** [review]
感觉之前好像没这些错误..辛苦再修改下吧

**Comment by LyndonKong** [comment]
Fix in our new commit

**Review by sunzhongkai588 — COMMENTED** [review]
看了下 ci不过原因，还是代码风格的问题，估计是 中英文or 中文和数字 之间没有加空格
image
- 建议用 [pre-commit](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/docs_contributing_guides_cn.html#pre-commit) 统一检查下，pre-commit能够自动修复该问题。 或者手动检查下两篇文档是否存在该类问题并修正。目前我已经看到几处 中英文 or 中文和数字 之间没有加空格。
- image

**Comment by LyndonKong** [comment]
> 看了下 ci不过原因，还是代码风格的问题，估计是 中英文or 中文和数字 之间没有加空格 image
> 
> * 建议用 [pre-commit](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/docs_contributing_guides_cn.html#pre-commit) 统一检查下，pre-commit能够自动修复该问题。 或者手动检查下两篇文档是否存在该类问题并修正。目前我已经看到几处 中英文 or 中文和数字 之间没有加空格。
> * image



> 看了下 ci不过原因，还是代码风格的问题，估计是 中英文or 中文和数字 之间没有加空格 image
> 
> * 建议用 [pre-commit](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/docs_contributing_guides_cn.html#pre-commit) 统一检查下，pre-commit能够自动修复该问题。 或者手动检查下两篇文档是否存在该类问题并修正。目前我已经看到几处 中英文 or 中文和数字 之间没有加空格。
> * image

Thanks for your kind suggesstions, we have run pre-commit in our new commit.

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Review by luotao1 — COMMENTED** [review]
请加上对float16/bfloat16数据类型的支持说明

**Comment by LyndonKong** [comment]
> 请加上对float16/bfloat16数据类型的支持说明

Sure, the support for float16 and bfloat16 is now added.

**Comment by LyndonKong** [comment]
This PR is ready to be reviewed.
